### PR TITLE
Adding optional encoding arg, and text_partition tests

### PR DESCRIPTION
**Pull request by amanda103**
Added optional `encoding` argument to the `partition_{html, email, text}`

Added to happy path test for text, and added one test for sad path. Hoping to get feedback on logic and testing before continuing down this path for html and email partitioning.

reference issue: https://github.com/Unstructured-IO/unstructured/issues/214

### CODE REVIEW

**Comment by tomaarsen** [comment]
Could you use `make tidy` to satisfy the Linting CI, and update the version in `CHANGELOG` and `unstructured/__version__.py`? It will help the continuous integration pass.
Thanks in advance!

**Review by cragwolfe — COMMENTED** [review]
Code changes look good! In addition to `make tidy` as @tomaarsen , please bump `unstructured/__version__.py` to the next `...-devN` version and update CHANGELOG.md. And for all CI checks to pass, the branch must also be brought up to date with main.

It also looks like PDF's can have their own additional encoding issues, but that looks beyond the scope of this PR. https://stackoverflow.com/questions/29467539/encoding-of-pdf-text-string

**Review by cragwolfe — APPROVED** [review]
Approving conditional on tidy + CHANGELOG changes in prior comment, which of course are needed to pass CI anyway.

File: test_unstructured/partition/test_text.py [hunk 1]
```diff
@@ -18,13 +18,22 @@
...
 
-def test_partition_text_from_filename():
-    filename = os.path.join(DIRECTORY, "..", "..", "example-docs", "fake-text.txt")
-    elements = partition_text(filename=filename)
+@pytest.mark.parametrize("filename,encoding", [('fake-text.txt', 'utf-8'), ('fake-text.txt', None), ('fake-text-utf-16-be.txt', 'utf-16-be')])
+def test_partition_text_from_filename(filename, encoding):
+    filename = os.path.join(DIRECTORY, "..", "..", "example-docs", filename)
+    elements = partition_text(filename=filename, encoding=encoding)
     assert len(elements) > 0
+    print([str(e) for e in elements])
```
Comment (tomaarsen) [inline test_unstructured/partition/test_text.py:26]:
```suggestion
```
Perhaps this was leftover from debugging? I think it's best to remove these kinds of`print` calls from tests.

Comment (amanda103) [reply test_unstructured/partition/test_text.py:26]: Good catch, yes definitely left over from debugging :)

**Pull request merged**